### PR TITLE
TEST/ASYNC: Add retries to test_async_timer_mt.multithread

### LINKED ISSUE #3010: failure in signal/test_async_timer_mt.multithread
```
08:54:09 [----------] 1 test from signal/test_async_timer_mt
08:54:09 [ RUN      ] signal/test_async_timer_mt.multithread/0
08:54:10 /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node-upstream/worker/1/contrib/../test/gtest/ucs/test_async.cc:726: Failure
08:54:10 Expected: (count) >= ((int)(COUNT * 0.10)), actual: 2 vs 4
08:54:10 [  FAILED  ] signal/test_async_timer_mt.multithread/0, where GetParam() = 0 (380 ms)
08:54:10 [----------] 1 test from signal/test_async_timer_mt (380 ms total)
```

### PR DESCRIPTION
**Pull request by yosefe**
Fixes #3010

### CODE REVIEW

**Comment by brminich** [comment]
env issues

bot:mlx:retest

File: src/uct/ib/mlx5/ib_mlx5.c [hunk 1]
```diff
@@ -115,7 +115,7 @@ void uct_ib_mlx5_check_completion(uct_ib_iface_t *iface, uct_ib_mlx5_cq_t *cq,
         /* update ci before invoking error callback, since it can poll on cq */
         UCS_STATIC_ASSERT(MLX5_CQE_REQ_ERR & (UCT_IB_MLX5_CQE_OP_OWN_ERR_MASK >> 4));
         ++cq->cq_ci;
-        status = uct_ib_mlx5_completion_with_err((void*)cqe, UCS_LOG_LEVEL_DEBUG);
+        status = uct_ib_mlx5_completion_with_err((void*)cqe, UCS_LOG_LEVEL_ERROR);
```
Comment (brminich) [inline src/uct/ib/mlx5/ib_mlx5.c]: how is it related?

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8175/ for details (Mellanox internal link).

File: src/uct/ib/mlx5/ib_mlx5.c (hunk 1 above)
Comment (yosefe) [reply src/uct/ib/mlx5/ib_mlx5.c]: committed by mistake

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8188/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8182/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8189/ for details (Mellanox internal link).

**Pull request merged**